src/Client.php

```diff
@@ -0,0 +1,163 @@
+            $method = strtoupper($method);
+        }
+        // if there are no headers but a body set header to json by default
+        if(!isset($headers['content-type']) && is_array($body) && count($body) > 0) {
+            $headers['content-type'] = 'application/json';
+        }
+        if(isset($headers['content-type'])) {
+            match ($headers['content-type']) { // Convert the body to the appropriate format
+                'application/json' => $body = json_encode($body),
+                'application/x-www-form-urlencoded' => $body = $this->flatten($body),
+                'multipart/form-data' => $body = $this->flatten($body),
+                'application/graphql' => $body = $body[0],
+                default => $body = $body,
+            };
+        }
+        if(!is_array($headers)) {
+            $headers = [];
+        }
+        foreach ($headers as $i => $header) { // convert headers to appropriate format
+            $headers[] = $i . ':' . $header;
+            unset($headers[$i]);
+        }
+        if($query) {  // if the request has a query string, append it to the request URI
+            $url .= '?' . http_build_query($query);
+        }
+        $resp_headers = [];
+        // Initialize the curl session
+        $ch = curl_init();
+        // Set the request URI
+        curl_setopt($ch, CURLOPT_URL, $url);
+        // Set the request headers
+        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+        // Set the request method
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
+        // Set the request body
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
+        // Save the response headers
+        curl_setopt($ch, CURLOPT_HEADERFUNCTION, function ($curl, $header) use (&$resp_headers) {
+            $len = strlen($header);
+            $header = explode(':', $header, 2);
+
+            if (count($header) < 2) { // ignore invalid headers
+                return $len;
+            }
+
+            $resp_headers[strtolower(trim($header[0]))] = trim($header[1]);
+            return $len;
+        });
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 0);
+        curl_setopt($ch, CURLOPT_TIMEOUT, 15);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        $resp_body = curl_exec($ch); // Execute the curl session
+        $resp_status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+        if (curl_errno($ch)) {
+            $error_msg = curl_error($ch);
+            print_r(curl_errno($ch));
+        }
+        curl_close($ch);
+
+        if (isset($error_msg)) {
+            // TODO - Handle cURL error accordingly
+            throw new FetchException($error_msg);
+        }
+        $resp = new Response(
+            method: $method,
+            url: $url,
+            statusCode: $resp_status,
+            headers: $resp_headers,
+            body: strval($resp_body),
```

Comment:
Im not sure if you are supposed to convert the response body to a string.. there are other kinds of responses that also involve sending binary data... so I dont think body should be stored as a string.. 

Please make sure your test cases handle this.

Reply:
okay, I am changing the body to `mixed` type and working on adding more testcases